Update Ember.js example to use Ember CLI

xtian opened the pull request:
Using Ember CLI is now the default method of building Ember apps, so I wrote a new implementation of TodoMVC using the latest versions of Ember, Ember Data, and Ember CLI. You can view it as its own repo here: [xtian/ember-cli-todomvc](https://github.com/xtian/ember-cli-todomvc).

However, since Ember CLI is a build system that runs a local server for development and produces compiled output for production deployment, it has a large number of command line and server dependencies that wouldn't make sense to commit into this repo. 

So I'd love feedback on the app and how best to fit it into this project. Would it be acceptable to just commit the compiled output in a `dist` folder?

Related issues:
* #1069
* #1343

arthurvr commented on examples/emberjs/.bowerrc:
```diff
@@ -0,0 +1,4 @@
+{
+  "directory": "bower_components",
```
isn't this default?

xtian replied:
It can be overwritten in `~/.bowerrc` so Ember CLI outputs this file with the default blueprint so it can guarantee the location.

arthurvr replied:
Ugh :cry: that's okay then I guess.

arthurvr commented on examples/emberjs/.editorconfig:
```diff
@@ -0,0 +1,22 @@
+indent_size = 2
+
+[*.md]
+trim_trailing_whitespace = false
```
Is there a reason we need an editorconfig in here? We have one in the todomvc root which all code should adhere to.

arthurvr commented:
@tastejs/todomvc Any volunteers to review this? I'm not really an Ember guy.

samccone commented:
Hey @xtian , first off thanks so much for doing this PR.

Would you mind committing the compiled output into dist (for now, we can decide if there is a better location for this later). 

Once that is done I will do a thorough review. Thanks!

samccone commented:
hey @xtian any progress on this front :)

xtian commented:
@samccone Sorry! I pushed up the compiled output.

samccone commented:
thanks @xtian :clap:

arthurvr commented:
> Once that is done I will do a thorough review. Thanks!

ping @samccone

samccone commented:
hey @xtian a few things:

1. Would you mind adding the src files here to https://github.com/tastejs/todomvc/blob/master/.jscsrc so that we do not try and lint them
2. rebase off of master 
3. (hopefully the tests pass and we can :ship: )

samccone commented:
friendly ping @xtian

xtian commented:
@samccone Alright, I updated the app to Ember 2.1.0 and added the files to the JSCS exclude. Let me know if there are any other changes you'd like me to make.

Thanks!

samccone commented:
Awesome we are almost there :racehorse: 

Since ember now is an "exception" for where the app lives, can you add it to the list here?
https://github.com/tastejs/todomvc/blob/master/tests/allTests.js#L35-L42

Thanks in advance :sparkles:

samccone commented:
Alright, well now it looks like there are just a few failing specs to work through.. Let me know if you think anything is amiss on the test side.

samccone commented:
friendly ping @xtian  :)

samccone commented:
Hey guys, I am going to close the PR, just ping us when you push and think it is good to go. Thanks

Also I think there is a ready to go implementation that could just be dropped in (ref https://github.com/tastejs/todomvc/issues/1333)

xtian commented:
Ok awesome
